sdk/database/helper/connutil/sql.go

```diff
@@ -55,16 +56,32 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 		return nil, fmt.Errorf("connection_url cannot be empty")
 	}
 
+	// Do not allow the username or password template pattern to be used as
+	// part of the user-supplied username or password
+	if c.DisableEscaping &&
```

Comment:
I wonder if we should always disallow this, whether disable_escaping is set or not. I believe that the template can't be properly interpolated if escaping is enabled due to curly brackets, but the error could surface as a potential typo to the user vs accepting that as the input. I'm fine with leaving this as-is too though.

Reply:
I think I agree with you. My only reservation would be causing a problem in the very-unlikely-but-not-impossible case someone actually configured their database already with these values in the username or password. @austingebauer When you get a chance to review as well, do you have an opinion on this matter?

Reply:
Hmm, I see why we'd want to disallow this in general. I also wonder if that behavior change would break some users. Seems unlikely but possible as you've mentioned. I'm okay leaving this as-is.

Reply:
Decided to go with always disallowing it, changed in commit https://github.com/hashicorp/vault/pull/13414/commits/927d4600bba9e8ff2bdeb44586f1fed26d8b4c98